### PR TITLE
[Hub Generated] Review request for Microsoft.Consumption to add version stable/2019-10-01

### PR DESCRIPTION
**Pull request by sri-**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/sri-/azure-rest-api-specs/reservationTransactions-Microsoft.Consumption-2019-10-01...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for AME onboarding.
- [x] [Validation tools](https://aka.ms/swaggertools) have been run on swagger spec(s) updated and errors have all been fixed with this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, [please view the FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories.
*	adding/removing APIs.
* adding/removing properties.
* adding/removing API-version.
* adding a new service in Azure.

*Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.*

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them.

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json [hunk 1]
```diff
@@ -3439,28 +3594,57 @@
     "ReservationTransaction": {
       "description": "Reservation transaction resource.",
       "type": "object",
+      "discriminator": "kind",
```
Comment (pilor) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json:3597]: >"discriminator": "kind", [](start = 6, length = 24)

these kinds of changes to an existing api-version will break clients.  Please introduce a new api-version

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json [hunk 2]
```diff
@@ -3439,28 +3594,57 @@
...
+          }
+        },
+        "nextLink": {
+          "description": "The link (url) to the next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "LegacyReservationTransactionsListResult": {
```
Comment (pilor) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json:3639]: >"LegacyReservationTransactionsListResult" [](start = 4, length = 41)

this isn't referenced

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json [hunk 3]
```diff
@@ -3439,28 +3594,57 @@
...
-        "properties": {
-          "x-ms-client-flatten": true,
-          "$ref": "#/definitions/ReservationTransactionProperties",
-          "title": "Reservation Transaction properties"
+        "kind": {
+          "type": "string",
+          "description": "Specifies the kind of reservation transaction.",
+          "enum": [
+            "Legacy",
+            "Modern"
```
Comment (pilor) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json:3609]: when does a legacy transaction get returned and when does a modern transaction get returned?

**Review by pilor — CHANGES_REQUESTED** [review]
:clock1:

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json (hunk 3 above)
Comment (sri-) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json:3609]: When the API is called with a Modern billing account, a modern transaction is returned. When it is called with a legacy Enrollment number, legacy reservation transaction is returned.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json (hunk 1 above)
Comment (sri-) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json:3597]: The API only supports legacy accounts currently. The only change for the existing callers is an additional "kind" property. 

With this change, callers with modern accounts will be able to use the API.

Adding a new api-Version is an expensive change on our side, which we are planning to do some time later.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json (hunk 2 above)
Comment (sri-) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json]: updated.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json (hunk 1 above)
Comment (pilor) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json:3597]: What happens when a user with a modern account tries to use this API with existing clients that don't support this new contract?

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/examples/ReservationTransactionsListByBillingAccountId.json [hunk 4]
```diff
@@ -1,67 +1,39 @@
 {
   "parameters": {
-    "api-version": "2019-10-01",
-    "billingAccountId": "123456",
-    "$filter": "properties/eventDate ge 2019-09-09 AND properties/eventDate le 2019-09-10"
+    "api-version": "2020-01-01",
```
Comment (NullMDR) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/examples/ReservationTransactionsListByBillingAccountId.json:3]: Why do you update the api-version in example file?

Comment (sri-) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/examples/ReservationTransactionsListByBillingAccountId.json]: That's a leftover from my previous attempt of adding a new api-Version. Updated it.

**Review by pilor — APPROVED** [review]

**Comment by NullMDR** [comment]
Hi @sri- you have a breaking change in this spec https://github.com/Azure/azure-rest-api-specs/blob/master/specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json#L3331 . Seems like it's a mistake?

**Comment by sri-** [comment]
> Hi @sri- you have a breaking change in this spec https://github.com/Azure/azure-rest-api-specs/blob/master/specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json#L3331 . Seems like it's a mistake in old spec? The default value looks like description

Yeah it is a bug in the old spec.

**Comment by NullMDR** [comment]
@akning-ms Please help to merge this PR with breaking change that fix the bug.

**Pull request merged**